core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -90,20 +150,23 @@ class AbstractTrainer:
     def __init__(
         self,
         path: str,
+        *,
         problem_type: str,
-        eval_metric=None,
-        num_classes=None,
-        quantile_levels=None,
-        low_memory=False,
-        feature_metadata=None,
-        k_fold=0,
-        n_repeats=1,
-        sample_weight=None,
-        weight_evaluation=False,
-        save_data=False,
-        random_state=0,
-        verbosity=2,
+        num_classes: int = None,
+        feature_metadata: FeatureMetadata = None,
+        eval_metric: Scorer = None,
+        quantile_levels: List[float] = None,
+        low_memory: bool = True,
```

Comment:
Note: low_memory was previous default False, however in all cases AbstractTrainer was initialized, it was set to True. False is not actively supported, so I've updated the default to True here, should have no impact.